tools/form-bridge/lambda_test.py

```diff
@@ -0,0 +1,305 @@
+
+        # Verify response structure
+        assert result['statusCode'] == 200
+
+        response_body = json.loads(result['body'])
+        assert response_body['message'] == 'Form submitted successfully'
+        assert 'zendesk_response' in response_body
+
+        # Verify send_to_zendesk was called with correct data
+        mock_send.assert_called_once()
+        call_args = mock_send.call_args[0][0]  # Get the ticket_data argument
+
+        assert call_args['ticket']['subject'] == 'Technical support for Thunderbird - Test Support Request'
+        assert call_args['ticket']['comment']['body'] == 'This is a test'
+        assert call_args['ticket']['requester']['name'] == 'Test User'
+        assert call_args['ticket']['requester']['email'] == 'test@example.com'
+        assert call_args['ticket']['brand_id'] == 43075490201747
+
+def test_lambda_handler():
+    """Test the full Lambda handler with normal body"""
+    _test_lambda_handler_helper(is_base64_encoded=False)
+
+def test_lambda_handler_base64_encoded():
+    """Test the full Lambda handler with base64 encoded body"""
+    _test_lambda_handler_helper(is_base64_encoded=True)
+
+def test_missing_fields():
+    """Test handling of missing form fields"""
+    # Form data with missing fields
+    form_data = {
+        'tfa_211': 'Support Request',
+        # Missing tfa_163 (comment)
+        'tfa_1': 'John Doe',
+        # Missing tfa_10 (email)
+    }
+
+    result = map_form_to_zendesk(form_data)
+
+    # Should use defaults for missing fields
+    assert result['ticket']['subject'] == 'Support Request'
+    assert result['ticket']['comment']['body'] == 'No comment provided'
+    assert result['ticket']['requester']['name'] == 'John Doe'
+    assert result['ticket']['requester']['email'] == ''
+
+
+
+def test_lambda_handler_with_empty_fields():
+    """Handler should map empty fields to defaults and succeed"""
+    with patch('lambda_function.send_to_zendesk') as mock_send:
+        mock_send.return_value = (True, {"ok": True})
+
+        body = (
+            "tfa_95=tfa_197"  # Technical support dropdown
+            "&tfa_211="       # empty subject
+            "&tfa_163="       # empty comment
+            "&tfa_1=%20%20"   # whitespace-only name
+            "&tfa_10="        # empty email
+        )
+
+        event = {
+            "body": body,
+            "isBase64Encoded": False,
+            "httpMethod": "POST",
+            "headers": {"content-type": "application/x-www-form-urlencoded"}
+        }
+
+        result = lambda_handler(event, None)
+        assert result['statusCode'] == 200
+
+        call_args = mock_send.call_args[0][0]
+        assert call_args['ticket']['subject'] == 'Technical support for Thunderbird - No subject provided'
+        assert call_args['ticket']['comment']['body'] == 'No comment provided'
+        assert call_args['ticket']['requester']['name'] == 'No name provided'
+        assert call_args['ticket']['requester']['email'] == ''
```

Comment:
While this is valid, do we want to error out here? I would assume we want at least their email address otherwise the submission is just spam.

Reply:
So, right now, what will happen is we pass the empty email but Zendesk API will fail and so we'll return a 500 on that basis.

I guess it's reasonable to fail before we even call Zendesk here, though I don't really want to try to duplicate all the potential Zendesk error conditions.

But in this case I agree there can't be a legitimate request with an empty email.